spec/rental_spec.rb

```diff
@@ -0,0 +1,34 @@
+require 'spec_helper'
+require 'date'
+require_relative '../rental'
+
+describe Rental do
+  before :each do
+    @book = Book.new('The Great Gatsby', 'F. Scott Fitzgerald')
+    @person = Person.new(30, 'John Doe', parent_permission: true)
+    @rental_date = Date.new(2023, 11, 1)
+    @rental = Rental.new(@rental_date, @book, @person)
+  end
+
+  describe '#new' do
+    it 'takes three parameters and returns a Rental object' do
+      expect(@rental).to be_an_instance_of(Rental)
+    end
+
+    it 'adds the rental to the book and person rentals lists' do
+      expect(@book.rentals).to include(@rental)
+      expect(@person.rentals).to include(@rental)
+    end
+  end
+
+  describe '#rental_to_json' do
+    it 'returns a hash representing the rental' do
+      expected_hash = {
+        date: @rental_date,
+        person: 'John Doe',
+        book: 'The Great Gatsby'
+      }
+      expect(@rental.rental_to_json).to eql(expected_hash)
+    end
```

Comment:
- [ ] @Team Great work adding tests for the rental class, however you forgot to add tests for these methods ` self.from_json, save_rental, load_rental`, its  project requirement to add tests for all methods
![image](https://github.com/Omwomo/school-library/assets/61408860/a1e0a748-ab1a-40c5-b631-9873c106c7df)